File: LeetCode/src/1441.js
var buildArray = function(target, n) {
    if (target[target.length-1] > n) return -1;
    
    const result = [];
    let element = 1;
    for (let t of target) {
        while (t >= element) {
            result.push('Push');
            (t != element) && result.push('Pop');
            element += 1;
        }
    }
    return result;
};

File: LeetCode/src/1441.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        result = list()
        element = 1
        for val in target:
            while element <= val:
                result.append('Push') 
                (element != val) and result.append('Pop')
                element += 1
        return result
